### PR TITLE
[dev-v2.4] Update k8s v1.18.8, v1.17.11, v1.16.14, v1.16.15

### PR DESCRIPTION
**Pull request by PennyScissors**
This PR includes:
- k8s update to v1.18.8, v1.17.11, v1.16.14, v1.16.15
- Corrected image sources to rancher custom builds and/or mirrors. This specifically addresses issue https://github.com/rancher/rancher/issues/27552 . This is unrelated to the k8s update, just piggybacking on this PR.

Related PR's:
https://github.com/rancher/hyperkube/pull/134
https://github.com/rancher/hyperkube/pull/135
https://github.com/rancher/hyperkube/pull/136
https://github.com/rancher/hyperkube/pull/137

Related Issues:
https://github.com/rancher/rancher/issues/28576
https://github.com/rancher/rancher/issues/27552

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -2651,6 +2651,41 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  m("rancher/kubelet-pause:v0.1.4"),
 			Nodelocal:                 m("k8s.gcr.io/k8s-dns-node-cache:1.15.7"),
 		},
+		// Enabled out of band for Rancher v2.4.4+
+		// Reminder: This template contains Nodelocal image which isn't in templates before v2.3.6
+		"v1.16.14-rancher1-1": {
+			Etcd:                      m("rancher/coreos-etcd:v3.3.15-rancher1"),
+			Kubernetes:                m("rancher/hyperkube:v1.16.14-rancher1"),
+			Alpine:                    m("rancher/rke-tools:v0.1.60"),
```
Comment (ryansann) [inline rke/k8s_rke_system_images.go:2659]: `v1.16.13-rancher1-2` had `rke-tools:v0.1.64`.. was it intentional to change this to an earlier version?

Comment (PennyScissors) [reply rke/k8s_rke_system_images.go:2659]: No, not sure how that happened. I'll get that updated, and verify the other ones.

Comment (ryansann) [reply rke/k8s_rke_system_images.go:2659]: Cool, looks like the other ones have the same issue

**Comment by PennyScissors** [comment]
@deniseschannon I messed something up when rebasing and the template you reviewed was slightly old. Please review the new ones which has different comments on it.

**Comment by deniseschannon** [comment]
> @deniseschannon I messed something up when rebasing and the template you reviewed was slightly old. Please review the new ones which has different comments on it.

It sill says "Enabled in v2.4.6" but should say "Enabled out of band post v2.4.6".

**Comment by PennyScissors** [comment]
> > @deniseschannon I messed something up when rebasing and the template you reviewed was slightly old. Please review the new ones which has different comments on it.
> 
> It still says "Enabled in v2.4.6" but should say "Enabled out of band post v2.4.6".

Updated comments

File: rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
-			PodInfraContainer:         m("gcr.io/google_containers/pause:3.1"),
+			PodInfraContainer:         m("rancher/pause:3.1"),
 			Ingress:                   m("rancher/nginx-ingress-controller:nginx-0.32.0-rancher1"),
-			IngressBackend:            m("k8s.gcr.io/defaultbackend:1.5-rancher1"),
-			MetricsServer:             m("gcr.io/google_containers/metrics-server:v0.3.4"),
-			CoreDNS:                   m("coredns/coredns:1.6.2"),
-			CoreDNSAutoscaler:         m("gcr.io/google_containers/cluster-proportional-autoscaler:1.7.1"),
+			IngressBackend:            m("rancher/nginx-ingress-controller-defaultbackend:1.5-rancher1"),
+			MetricsServer:             m("rancher/metrics-server:v0.3.4"),
+			CoreDNS:                   m("rancher/coredns:1.6.2"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2684]: This is not the right image: https://hub.docker.com/repository/docker/rancher/coredns/tags?page=1

This one shouldn't be changed.

File: rke/k8s_rke_system_images.go [hunk 3]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
-			KubeDNS:                   m("gcr.io/google_containers/k8s-dns-kube-dns:1.15.0"),
-			DNSmasq:                   m("gcr.io/google_containers/k8s-dns-dnsmasq-nanny:1.15.0"),
-			KubeDNSSidecar:            m("gcr.io/google_containers/k8s-dns-sidecar:1.15.0"),
-			KubeDNSAutoscaler:         m("gcr.io/google_containers/cluster-proportional-autoscaler:1.7.1"),
-			Flannel:                   m("quay.io/coreos/flannel:v0.12.0"),
+			KubeDNS:                   m("rancher/k8s-dns-kube-dns:1.15.0"),
+			DNSmasq:                   m("rancher/k8s-dns-dnsmasq-nanny:1.15.0"),
+			KubeDNSSidecar:            m("rancher/k8s-dns-sidecar:1.15.0"),
+			KubeDNSAutoscaler:         m("rancher/cluster-proportional-autoscaler:1.7.1"),
+			Flannel:                   m("rancher/coreos-flannel:v0.12.0"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2667]: This image `quay.io/coreos/flannel:v0.12.0` exists and shouldn't be updated.

File: rke/k8s_rke_system_images.go [hunk 4]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
-			CalicoNode:                m("quay.io/calico/node:v3.13.4"),
-			CalicoCNI:                 m("quay.io/calico/cni:v3.13.4"),
-			CalicoControllers:         m("quay.io/calico/kube-controllers:v3.13.4"),
-			CalicoCtl:                 m("quay.io/calico/ctl:v3.13.4"),
-			CalicoFlexVol:             m("quay.io/calico/pod2daemon-flexvol:v3.13.4"),
-			CanalNode:                 m("quay.io/calico/node:v3.13.4"),
-			CanalCNI:                  m("quay.io/calico/cni:v3.13.4"),
-			CanalFlannel:              m("quay.io/coreos/flannel:v0.12.0"),
-			CanalFlexVol:              m("quay.io/calico/pod2daemon-flexvol:v3.13.4"),
+			CalicoNode:                m("rancher/calico-node:v3.13.4"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2669]: This image `quay.io/calico/node:v3.13.4` and all the calico and canal images exist and shouldn't be updated.

File: rke/k8s_rke_system_images.go [hunk 5]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			CalicoCtl:                 m("rancher/calico-ctl:v3.13.4"),
+			CalicoFlexVol:             m("rancher/calico-pod2daemon-flexvol:v3.13.4"),
+			CanalNode:                 m("rancher/calico-node:v3.13.4"),
+			CanalCNI:                  m("rancher/calico-cni:v3.13.4"),
+			CanalFlannel:              m("rancher/coreos-flannel:v0.12.0"),
+			CanalFlexVol:              m("rancher/calico-pod2daemon-flexvol:v3.13.4"),
 			WeaveNode:                 m("weaveworks/weave-kube:2.6.4"),
 			WeaveCNI:                  m("weaveworks/weave-npc:2.6.4"),
-			PodInfraContainer:         m("gcr.io/google_containers/pause:3.1"),
+			PodInfraContainer:         m("rancher/pause:3.1"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2680]: This image `gcr.io/google_containers/pause:3.1` exists and shouldn't be updated.

File: rke/k8s_rke_system_images.go [hunk 6]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
 			Ingress:                   m("rancher/nginx-ingress-controller:nginx-0.32.0-rancher1"),
-			IngressBackend:            m("k8s.gcr.io/defaultbackend:1.5-rancher1"),
-			MetricsServer:             m("gcr.io/google_containers/metrics-server:v0.3.4"),
-			CoreDNS:                   m("coredns/coredns:1.6.2"),
-			CoreDNSAutoscaler:         m("gcr.io/google_containers/cluster-proportional-autoscaler:1.7.1"),
+			IngressBackend:            m("rancher/nginx-ingress-controller-defaultbackend:1.5-rancher1"),
+			MetricsServer:             m("rancher/metrics-server:v0.3.4"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2683]: This image `gcr.io/google_containers/metrics-server:v0.3.4` exists and shouldn't be updated.

File: rke/k8s_rke_system_images.go [hunk 7]
```diff
@@ -2654,37 +2654,37 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			CoreDNS:                   m("rancher/coredns:1.6.2"),
+			CoreDNSAutoscaler:         m("rancher/cluster-proportional-autoscaler:1.7.1"),
 			WindowsPodInfraContainer:  m("rancher/kubelet-pause:v0.1.4"),
-			Nodelocal:                 m("k8s.gcr.io/k8s-dns-node-cache:1.15.7"),
+			Nodelocal:                 m("rancher/k8s-dns-node-cache:1.15.7"),
```
Comment (deniseschannon) [inline rke/k8s_rke_system_images.go:2687]: This image `k8s.gcr.io/k8s-dns-node-cache:1.15.7` exists and shouldn't be changed.

**Comment by PennyScissors** [comment]
@deniseschannon I reverted the images. Only kept changes for these:
```
"quay.io/coreos/etcd:v3.4.3-rancher1"
"gcr.io/google_containers/cluster-proportional-autoscaler:1.7.1"
"k8s.gcr.io/defaultbackend:1.5-rancher1"
```

**Review by deniseschannon — DISMISSED** [review]
This looks good but since v2.4.6 is imminent. The comments should be updated to post v2.4.7 😭

**Comment by PennyScissors** [comment]
@deniseschannon No problem, I'll get those updated.

**Comment by PennyScissors** [comment]
Added v1.16.15

**Review by kinarashah — APPROVED** [review]

**Comment by kinarashah** [comment]
@PennyScissors squash 2nd and 3rd commit, and good to go

**Review by kinarashah — APPROVED** [review]

**Pull request merged**